### DISCUSSION TITLE
Request: add additional progress indicators

### DISCUSSION THREAD

**Discussion by Fiwx** [PGScatalog/pgsc_calc#233] [Q&A] [answered]
### Description of feature

Sometimes, for various reasons (a bug, an ssh connection silently being dropped, etc.), pgsc_calc freezes or otherwise gets stuck in the "checkboxes" stage. However, in normal use (especially when running multiple or large scores), there can be large periods of time where pgsc_calc is correctly progressing through the checkboxes, but simply takes a long time to "move" (i.e., check another box or finish). This means that there can be long periods of time where I'm not sure if I need to restart the program or not. It would be nice to have a verbose progress option which just gives more frequent updates about what it is doing, or some other indicator that it is still progressing and not frozen.

Not a necessity, but if it's easy to implement, this would be convenient!

**Comment by smlmbrt** [comment, marked as answer]
Hi @Fiwx, the displaying of the actual progress (checkboxes) is handled by  nextflow itself. One thing it does is list the folder where the work is happening, example:
```
[49/d28766] process > PGSC_CALC:PGSCALC:INPUT_CHECK:SAMPLESHEET_JSON (samplesheet.csv)           [100%] 1 of 1 ✔
[c3/a8e0d9] process > PGSC_CALC:PGSCALC:INPUT_CHECK:SCOREFILE_CHECK                              [100%] 1 of 1 ✔
[-        ] process > PGSC_CALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_VCF                               -
[7c/5cca6c] process > PGSC_CALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_BFILE (cineca_synthetic_subset)   [100%] 1 of 1 ✔
[3b/ce0e39] process > PGSC_CALC:PGSCALC:MAKE_COMPATIBLE:MATCH_VARIANTS (cineca_synthetic_subset) [100%] 1 of 1 ✔
[2e/fb3233] process > PGSC_CALC:PGSCALC:APPLY_SCORE:PLINK2_SCORE (cineca_synthetic_subset)       [100%] 1 of 1 ✔
[b5/fc5b1e] process > PGSC_CALC:PGSCALC:APPLY_SCORE:SCORE_REPORT (1)                             [100%] 1 of 1 ✔
[03/009cb6] process > PGSC_CALC:PGSCALC:DUMPSOFTWAREVERSIONS (1)                                 [100%] 1 of 1 ✔
-[pgscatalog/pgsc_calc] Pipeline completed successfully-
```
If you wanted to check on the status of SCORE_REPORT while it's in progress you can look at the log file by opening a second terminal and looking at the `work/b5/fc5b1e*/.command.log` file. Most of the software in pgsc_calc will output informative messages as it goes.
